build failing

@claude Run npx --no-install next build
⚠ No build cache found. Please configure build caching for faster rebuilds. Read more: https://nextjs.org/docs/messages/no-cache
Attention: Next.js now collects completely anonymous telemetry regarding usage.
This information is used to shape Next.js' roadmap and prioritize features.
You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
https://nextjs.org/telemetry

  ▲ Next.js 14.2.13

   Creating an optimized production build ...
Error: R] No matching export in "../../src/lib/number-formatters.ts" for import "createAutoScaledFormatter"

    ../../src/components/analyses/energiekaart-premies/EnergiekaartChart.tsx:16:9:
      16 │ import { createAutoScaledFormatter, formatNumber, formatCurrency }...
         ╵          ~~~~~~~~~~~~~~~~~~~~~~~~~

Error: R] No matching export in "../../src/lib/number-formatters.ts" for import "createAutoScaledFormatter"

    ../../src/components/analyses/shared/FilterableChart.tsx:6:9:
      6 │ import { createAutoScaledFormatter } from "@/lib/number-formatters"
        ╵          ~~~~~~~~~~~~~~~~~~~~~~~~~

Error: R] No matching export in "../../src/lib/number-formatters.ts" for import "createAutoScaledFormatter"

    ../../src/components/analyses/gebouwenpark/GebouwenChart.tsx:5:9:
      5 │ import { createAutoScaledFormatter, formatNumber } from "@/lib/numb...
        ╵          ~~~~~~~~~~~~~~~~~~~~~~~~~

Error: R] No matching export in "../../src/lib/number-formatters.ts" for import "createAutoScaledFormatter"

    ../../src/components/analyses/shared/FilterableChart.tsx:6:9:
      6 │ import { createAutoScaledFormatter } from "@/lib/number-formatters"
        ╵          ~~~~~~~~~~~~~~~~~~~~~~~~~

Error: R] No matching export in "../../src/lib/number-formatters.ts" for import "createAutoScaledFormatter"

    ../../src/components/analyses/gemeentelijke-investeringen/InvesteringenBVSection.tsx:24:2:
      24 │   createAutoScaledFormatter,
         ╵   ~~~~~~~~~~~~~~~~~~~~~~~~~

Error: R] No matching export in "../../src/lib/number-formatters.ts" for import "getScaledLabel"

    ../../src/components/analyses/gemeentelijke-investeringen/InvesteringenBVSection.tsx:25:2:
      25 │   getScaledLabel,
         ╵   ~~~~~~~~~~~~~~

Error: R] No matching export in "../../src/lib/number-formatters.ts" for import "createAutoScaledFormatter"

    ../../src/components/analyses/gemeentelijke-investeringen/InvesteringenREKSection.tsx:24:2:
      24 │   createAutoScaledFormatter,
         ╵   ~~~~~~~~~~~~~~~~~~~~~~~~~

Error: R] No matching export in "../../src/lib/number-formatters.ts" for import "getScaledLabel"

    ../../src/components/analyses/gemeentelijke-investeringen/InvesteringenREKSection.tsx:25:2:
      25 │   getScaledLabel,
         ╵   ~~~~~~~~~~~~~~

Error: R] No matching export in "../../src/lib/number-formatters.ts" for import "createAutoScaledFormatter"

    ../../src/components/analyses/shared/FilterableChart.tsx:6:9:
      6 │ import { createAutoScaledFormatter } from "@/lib/number-formatters"
        ╵          ~~~~~~~~~~~~~~~~~~~~~~~~~

Error: R] No matching export in "../../src/lib/number-formatters.ts" for import "createAutoScaledFormatter"

    ../../src/components/analyses/shared/FilterableChart.tsx:6:9:
      6 │ import { createAutoScaledFormatter } from "@/lib/number-formatters"
        ╵          ~~~~~~~~~~~~~~~~~~~~~~~~~

Error: R] No matching export in "../../src/lib/number-formatters.ts" for import "createAutoScaledFormatter"

    ../../src/components/analyses/shared/FilterableChart.tsx:6:9:
      6 │ import { createAutoScaledFormatter } from "@/lib/number-formatters"
        ╵          ~~~~~~~~~~~~~~~~~~~~~~~~~

Error: R] No matching export in "../../src/lib/number-formatters.ts" for import "createAutoScaledFormatter"

    ../../src/components/analyses/shared/FilterableChart.tsx:6:9:
      6 │ import { createAutoScaledFormatter } from "@/lib/number-formatters"
        ╵          ~~~~~~~~~~~~~~~~~~~~~~~~~

Error: Found 9 problems in 12 documents.

 └── Encountered unexpected errors while processing of 9 documents. This is possibly a bug in Contentlayer. Please open an issue.
     
     • "bouwondernemers/content.mdx": UnexpectedMDXError: Error: Build failed with 1 error:
     ../../src/components/analyses/shared/FilterableChart.tsx:6:9: ERROR: No matching export in "../../src/lib/number-formatters.ts" for import "createAutoScaledFormatter"
     • "energiekaart-premies/content.mdx": UnexpectedMDXError: Error: Build failed with 1 error:
     ../../src/components/analyses/energiekaart-premies/EnergiekaartChart.tsx:16:9: ERROR: No matching export in "../../src/lib/number-formatters.ts" for import "createAutoScaledFormatter"
     • "faillissementen/content.mdx": UnexpectedMDXError: Error: Build failed with 1 error:
     ../../src/components/analyses/shared/FilterableChart.tsx:6:9: ERROR: No matching export in "../../src/lib/number-formatters.ts" for import "createAutoScaledFormatter"
     • "gebouwenpark/content.mdx": UnexpectedMDXError: Error: Build failed with 1 error:
     ../../src/components/analyses/gebouwenpark/GebouwenChart.tsx:5:9: ERROR: No matching export in "../../src/lib/number-formatters.ts" for import "createAutoScaledFormatter"
     • "gemeentelijke-investeringen/content.mdx": UnexpectedMDXError: Error: Build failed with 4 errors:
     ../../src/components/analyses/gemeentelijke-investeringen/InvesteringenBVSection.tsx:24:2: ERROR: No matching export in "../../src/lib/number-formatters.ts" for import "createAutoScaledFormatter"
     ../../src/components/analyses/gemeentelijke-investeringen/InvesteringenBVSection.tsx:25:2: ERROR: No matching export in "../../src/lib/number-formatters.ts" for import "getScaledLabel"
     ../../src/components/analyses/gemeentelijke-investeringen/InvesteringenREKSection.tsx:24:2: ERROR: No matching export in "../../src/lib/number-formatters.ts" for import "createAutoScaledFormatter"
     ../../src/components/analyses/gemeentelijke-investeringen/InvesteringenREKSection.tsx:25:2: ERROR: No matching export in "../../src/lib/number-formatters.ts" for import "getScaledLabel"
     • "huishoudensgroei/content.mdx": UnexpectedMDXError: Error: Build failed with 1 error:
     ../../src/components/analyses/shared/FilterableChart.tsx:6:9: ERROR: No matching export in "../../src/lib/number-formatters.ts" for import "createAutoScaledFormatter"
     • "starters-stoppers/content.mdx": UnexpectedMDXError: Error: Build failed with 1 error:
     ../../src/components/analyses/shared/FilterableChart.tsx:6:9: ERROR: No matching export in "../../src/lib/number-formatters.ts" for import "createAutoScaledFormatter"
     • "vastgoed-verkopen/content.mdx": UnexpectedMDXError: Error: Build failed with 1 error:
     ../../src/components/analyses/shared/FilterableChart.tsx:6:9: ERROR: No matching export in "../../src/lib/number-formatters.ts" for import "createAutoScaledFormatter"
     • "vergunningen-goedkeuringen/content.mdx": UnexpectedMDXError: Error: Build failed with 1 error:
     ../../src/components/analyses/shared/FilterableChart.tsx:6:9: ERROR: No matching export in "../../src/lib/number-formatters.ts" for import "createAutoScaledFormatter"
     

Error: Process completed with exit code 1.

--- a/embuild-analyses/src/lib/number-formatters.ts
+++ b/embuild-analyses/src/lib/number-formatters.ts
@@ -26,40 +26,76 @@ export function formatCurrency(value: number, decimals: number = 0): string {
 
 /**
  * Create an auto-scaled formatter for chart axes to prevent label overflow
+ *
+ * Determines the appropriate scale (B/M/K) based on the maximum absolute value
+ * in the data and returns a formatter function that applies Belgian locale formatting.
+ *
  * @param values - Array of numbers to determine the scale
  * @param isCurrency - Whether to format as currency (with € symbol)
- * @returns Object with formatter function and scale string
+ * @returns Object with:
+ *   - formatter: Function to format axis tick values with Belgian locale (space separator, comma decimal)
+ *   - scale: Scale suffix string ("B", "M", "K", or "" for no scaling)
  */
 export function createAutoScaledFormatter(
   values: number[],
   isCurrency: boolean = false
 ): { formatter: (value: number) => string; scale: string } {
-  const maxValue = Math.max(...values.filter((v) => !isNaN(v) && isFinite(v)))
+  // Filter out invalid values
+  const filteredValues = values.filter((v) => !isNaN(v) && isFinite(v))
 
-  if (maxValue >= 1_000_000_000) {
+  // Handle empty or all-invalid data
+  if (filteredValues.length === 0) {
+    return {
+      formatter: (value: number) => {
+        const formatted = value.toLocaleString("nl-BE", {
+          minimumFractionDigits: 0,
+          maximumFractionDigits: 0,
+        })
+        return isCurrency ? `€${formatted}` : formatted
+      },
+      scale: "",
+    }
+  }
+
+  // Use absolute max to handle negative values correctly
+  const maxAbsValue = Math.max(...filteredValues.map((v) => Math.abs(v)))
+
+  if (maxAbsValue >= 1_000_000_000) {
     // Billion scale
     return {
       formatter: (value: number) => {
         const scaled = value / 1_000_000_000
-        return isCurrency ? `€${scaled.toFixed(1)}` : scaled.toFixed(1)
+        const formatted = scaled.toLocaleString("nl-BE", {
+          minimumFractionDigits: 1,
+          maximumFractionDigits: 1,
+        })
+        return isCurrency ? `€${formatted}` : formatted
       },
       scale: "B",
     }
-  } else if (maxValue >= 1_000_000) {
+  } else if (maxAbsValue >= 1_000_000) {
     // Million scale
     return {
       formatter: (value: number) => {
         const scaled = value / 1_000_000
-        return isCurrency ? `€${scaled.toFixed(1)}` : scaled.toFixed(1)
+        const formatted = scaled.toLocaleString("nl-BE", {
+          minimumFractionDigits: 1,
+          maximumFractionDigits: 1,
+        })
+        return isCurrency ? `€${formatted}` : formatted
       },
       scale: "M",
     }
-  } else if (maxValue >= 10_000) {
+  } else if (maxAbsValue >= 10_000) {
     // Thousand scale (only if > 10k to avoid unnecessary scaling)
     return {
       formatter: (value: number) => {
         const scaled = value / 1_000
-        return isCurrency ? `€${scaled.toFixed(1)}` : scaled.toFixed(1)
+        const formatted = scaled.toLocaleString("nl-BE", {
+          minimumFractionDigits: 1,
+          maximumFractionDigits: 1,
+        })
+        return isCurrency ? `€${formatted}` : formatted
       },
       scale: "K",
     }
@@ -68,7 +104,11 @@ export function createAutoScaledFormatter(
   // No scaling needed
   return {
     formatter: (value: number) => {
-      return isCurrency ? `€${value.toFixed(0)}` : value.toFixed(0)
+      const formatted = value.toLocaleString("nl-BE", {
+        minimumFractionDigits: 0,
+        maximumFractionDigits: 0,
+      })
+      return isCurrency ? `€${formatted}` : formatted
     },
     scale: "",
   }

